docs/proposals/cert/Self-Signed_Certificate_Content_Standardization.md

```diff
@@ -0,0 +1,501 @@
+
+Definition of server and client components:
+- If a component has behavior of connecting to other components, it can be considered a client component
+- If a component has behavior of providing API endpoints, it can be considered a server component
+- The deployment scripts in `hack` and the mounted certificates in `artifacts` can be used to confirm related components
+
+- **8 Server Components:**
+  1. karmada-apiserver (acts as both server and client)
+  2. karmada-aggregated-apiserver (acts as both server and client)
+  3. karmada-scheduler-estimator (acts as both server and client)
+  4. karmada-metrics-adapter (acts as both server and client)
+  5. karmada-webhook (acts as both server and client)
+  6. karmada-search (acts as both server and client)
+  7. etcd
+  8. karmada-interpreter-webhook (acts as both server and client)
+
+- **11 Client Components:**
+  1. karmada-apiserver (connects to etcd)
+  2. karmada-aggregated-apiserver (connects to etcd, apiserver)
+  3. karmada-scheduler-estimator (connects to apiserver)
+  4. karmada-controller-manager (connects to apiserver)
+  5. karmada-metrics-adapter (connects to apiserver)
+  6. karmada-webhook (connects to apiserver)
+  7. karmada-search (connects to apiserver, etcd)
+  8. karmada-descheduler (connects to apiserver, karmada-scheduler-estimator)
+  9. karmada-scheduler (connects to apiserver, karmada-scheduler-estimator)
+  10. karmada-interpreter-webhook (connects to apiserver)
+  11. karmadactl (connects to apiserver)
+  12. karmada-agent (connects to apiserver)
+
+Current certificate allocation:
+
+- **Server Components:**
+  - Only apiserver and local etcd components have independent certificates
+  - All other server components share the same certificate with `cn=karmda-admin`
+  - Most components load certificates into containers through volume mounting
+
+- **Client Components:**
+  - All client components connecting to apiserver share the client certificate generated by karmada-apiserver, specifically for client interaction
+  - All client components connecting to etcd share the etcd-client certificate generated by etcd
+  - All client components connecting to karmada-scheduler-estimator use the gRPC verification certificate issued by karmada-scheduler-estimator
+
+Special Cases:
+1. For the two client components connecting to karmada-scheduler-estimator (karmada-scheduler and karmada-descheduler), since they use the gRPC protocol when connecting to karmada-scheduler-estimator, and they are the only components using gRPC protocol for this connection, the certificate is only used for TLS connection and does not contain identity verification attributes.
+2. Regarding the karmada-interpreter-webhook component, since it is only used for cluster e2e testing, it is considered a test component and does not play a role in actual production clusters. Therefore, it is unnecessary to change its certificate to a more granular one.
```

Comment:
This component is not only used by E2E but also a demo component. Users can refer to this component to implement their own components and apply them to the production environment.